test/models/test_transformers.py

```diff
@@ -0,0 +1,52 @@
+import torch
+import torchtext
+
+from ..common.parameterized_utils import nested_params
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+class TestTransformers(TorchtextTestCase):
+    @nested_params(
+        [True, False],
+        [True, False],
+    )
+    def test_padded_input_inference(self, with_no_grad, return_all_layers):
+        """test transformerencoder inference same with and without padding"""
+        from torchtext.models import RobertaEncoderConf, RobertaModel
+
+        def encoder_inference(encoder, input_lst, with_no_grad):
+            if(with_no_grad):
+                with torch.no_grad():
+                    res = [encoder(eval_input) for eval_input in input_lst]
+            else:
+                res = [encoder(eval_input) for eval_input in input_lst]
+            return res
+
+        # Roberta config except for less layers (2 instead of 12)
+        pad_idx = 1
+        encoder_conf = RobertaEncoderConf(
+            vocab_size=250002, 
+            embedding_dim=768, 
+            ffn_dimension=3072, 
+            padding_idx = pad_idx,
+            max_seq_len = 514,
+            num_attention_heads=12, 
+            num_encoder_layers=2,
+            dropout=0.1,
+            scaling=None,
+            normalize_before=False,
+        )
+        model = RobertaModel(encoder_conf)
+        model = model.eval()
+        # TODO: make return_all_layers a property of RobertaEncoderConf so it can be passed as arg
```

Comment:
Currently, return_all_layers cannot be set from RobertaEncoderConf because it's not a property of the config. Thus we need to manually set it below. Not a big deal right now, but something to consider.